### PR TITLE
swaps: osdnk review

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes TEAM2-255 TEAM2-161 TEAM2-256

## What changed + POW (plus any additional context for devs)
0x icon needed to be updated in 2 places not one:
https://cloud.skylarbarrera.com/Screen-Shot-2022-07-15-23-22-58.73.png

I slowed down the sheet animation by half, it looks a lot smoother now
before: https://cloud.skylarbarrera.com/Image-from-iOS-3-.MP4
after: https://cloud.skylarbarrera.com/Screen-Recording-2022-07-16-01-00-19.mp4

I moved the network check for new selected currencies closer to the actual event which helped fix keyboard handling, I also fixed a smaller bug where `hasShownWarning` wasn't updating until the next session
PoW: https://cloud.skylarbarrera.com/Screen-Recording-2022-07-17-23-01-32.mp4

## What to test

check 0x icon in swap settings 

play with swap details expanded state transition 

selecting new assets should work as expected, 
tokens on new networks should clear the other input after being selected 
we should only show the explainer once

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

**Comment by osdnk** [comment]
"selecting new assets should work as expected, new networks should clear the other and also only show the explainer once" i don't understand this part

File: src/hooks/useSwapCurrencyHandlers.js [hunk 1]
```diff
@@ -244,55 +202,27 @@ export default function useSwapCurrencyHandlers({
...
-        delayNext();
-        navigate(Routes.CURRENCY_SELECT_SCREEN, {
-          callback: inputFieldRef?.current?.clear,
-          chainId,
-          onSelectCurrency: updateInputCurrency,
-          restoreFocusOnSwapModal: () => setParams({ focused: true }),
-          title,
-          type: CurrencySelectionTypes.input,
-        });
+      dangerouslyGetParent().dangerouslyGetState().index = 0;
```
Comment (estebanmino) [inline src/hooks/useSwapCurrencyHandlers.js:216]: why `InteractionManager.runAfterInteractions` was removed?

Comment (skylarbarrera) [reply src/hooks/useSwapCurrencyHandlers.js:216]: accident, adding back

**Comment by skylarbarrera** [comment]
> "selecting new assets should work as expected, new networks should clear the other and also only show the explainer once" i don't understand this part

re-wrote but tbh ur good on doing a full review of this since you're lacking context

**Review by osdnk — APPROVED** [review]
i read the code and looks ok, but I'm too lazy to check if this works, so if you need extra dev QA, ping me

**Pull request merged**